VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -108,7 +172,7 @@ String getUserEnrolledCoursesURL(@NonNull String username) {
 
     public
     @NonNull
-    List<EnrolledCoursesResponse> getUserEnrolledCourses(@NonNull String username, boolean tryCache) throws HttpException {
+    List<EnrolledCoursesResponse> getUserEnrolledCourses(@NonNull String username, boolean tryCache) throws Exception {
```

Comment:
Unfortunately, the Retrofit 2 call executor can throw an unchecked exception if there is an error during the request creation or response processing stage. It has also removed the `RetrofitError` wrapper and global exception converter support that was present in version 1. Therefore, the caller needs to be prepared to catch any exception.

I suppose the only unchecked exception that we need to be concerned about at the moment would be `JsonSyntaxException`, which can be thrown by the Gson convertor upon encountering an error during conversion. We could catch that and rethrow it as a checked exception. However, since we're not handling these specific exceptions separately anywhere where this method is being called from (as they were originally using the equivalent method from the old `Api` class, which was also declared to throw the generic exception type), it might be safer to just leave it as it is for the moment.